source/read-write-pref.txt

```diff
@@ -0,0 +1,228 @@
+query. You can also customize how the server evaluates members. 
+
+For more detailed API documentation, see the `Read Preference API
+documentation <https://www.mongodb.com/docs/ruby-driver/current/api/Mongo/Collection/View/Readable.html#read_preference-instance_method>`__.
+
+.. list-table::
+   :widths: 25 25 50
+   :header-rows: 1
+
+   * - Parameter
+     - Type
+     - Description
+
+   * - ``mode``
+     - ``Symbol``
+     - Specifies a requirement or preference for which replica set
+       member the server reads from. The default mode, ``:primary``, specifies that
+       operations read from the primary member of the replica set.
+
+   * - ``tags`` *(optional)*
+     - ``Array<Hash>``
+     - Assigns tags to secondary replica set members to customize how the server evaluates
+       them. Tags cannot be used with the ``:primary`` read preference mode setting.
+
+   * - ``options`` *(optional)*
+     - ``Hash``
+     - Sets various options, including :manual:`hedge </core/read-preference-hedge-option/>`
+       and :manual:`maxStalenessSeconds </core/read-preference-staleness/>` that can be
+       applied to your read preference.
+
+Example: Set Read Preference and Concerns for a Transaction
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This code sets the read preference, read concern, and write concern for the operations in
+a transaction:
+
+.. literalinclude:: /includes/usage-examples/read-write-pref.rb
+   :language: ruby
+   :dedent:
+   :start-after: start-read-preference
+   :end-before: end-read-preference
+
+For more information about transactions, see `ruby-transactions <https://www.mongodb.com/docs/ruby-driver/upcoming/reference/transactions/>`_.
+
+Example: Set the Read Preference of a Cluster in the Connection String
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This code example creates a MongoClient that uses the "secondary" read preference mode
+when performing queries on a cluster:
+
+.. literalinclude:: /includes/usage-examples/read-write-pref.rb
+   :language: ruby
+   :dedent:
+   :start-after: start-read-preference-cluster
+   :end-before: end-read-preference-cluster
+
+This example also sets the ``maxStalenessSeconds`` option. For more information 
+about connection string options, see the :manual:`Connection String Options </reference/connection-string/#connection-string-options>`
+section in the manual.
+
+API Documentation
+-----------------
+
+To learn more about the methods and types mentioned in this guide, see the following API
+documentation:
+
+- `API Write Concern <https://www.mongodb.com/docs/ruby-driver/current/api/Mongo/Client.html#write_concern-instance_method>`__
+- `API Read Concern <{+api+}/Mongo/Client.html#read_concern-instance_method>`__
+- `API Read Preference <https://www.mongodb.com/docs/ruby-driver/current/api/Mongo/Collection/View/Readable.html#read_preference-instance_method>`__
```

Comment:
S: I don't think API is needed in these links since they're introduced as links to API pages
```suggestion
- `Write Concern <https://www.mongodb.com/docs/ruby-driver/current/api/Mongo/Client.html#write_concern-instance_method>`__
- `Read Concern <{+api+}/Mongo/Client.html#read_concern-instance_method>`__
- `Read Preference <https://www.mongodb.com/docs/ruby-driver/current/api/Mongo/Collection/View/Readable.html#read_preference-instance_method>`__
```